src/main/java/uk/gov/hmcts/ccd/data/casedetails/search/builder/GrantTypeQueryBuilder.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.ccd.data.casedetails.search.builder;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang3.StringUtils;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+
+public interface GrantTypeQueryBuilder {
+
+    String QUERY_WRAPPER = "( %s )";
+
+    String QUERY = "%s in (:classifications)";
+
+    String EMPTY = "";
+
+    String SECURITY_CLASSIFICATION = "security_classification";
+
+    String JURISDICTION = "jurisdiction";
+
+    String CASE_ID = "case_id";
+
+    String CASE_TYPE_ID = "case_type_id";
+
+    String LOCATION = "location";
+
+    String REGION = "region";
+
+    String createQuery(List<RoleAssignment> roleAssignments, Map<String, Object> params);
```

Comment:
Could we use Monad Optional in the all method signatures that can return value or NonValue ? then ifPresent or Map . Only if it is not a big change.